File: lib/player.dart
class Player {
  final String id;
  final String name;
  final String dorsal;
  final String? foto;
  final String? posicionPrincipal;
  int matchesPlayed;
  Map<String, int> stats;

  Player({
    required this.id,
    required this.name,
    required this.dorsal,
    this.foto,
    this.posicionPrincipal,
    this.matchesPlayed = 0,
    required this.stats,
  });

  // Helper to get goals quickly
  int get goals => stats['Gol'] ?? 0;
}

File: api/src/main/java/es/sl1iickdev/loloapi/models/Player.java
package es.sl1iickdev.loloapi.models;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "players")
public class Player {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Relación Muchos a Uno: Muchos jugadores pertenecen a un Equipo
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id") // Nombre de la columna FK en MariaDB
    private Team team;

    private String nombre;

    private int dorsal;

    @Column(name = "posicion_principal")
    private String posicionPrincipal;

    @Column(name = "tarjetas_acumuladas")
    private int tarjetasAcumuladas;

    private String foto;
}